kajas/backend/controllers/authController.js

```diff
@@ -0,0 +1,127 @@
+const bcrypt = require("bcryptjs");
+const User = require("../models/User");
+const UserInformation = require("../models/UserInformation");
+const db = require("../config/db");
+
+const signup = async (req, res) => {
+  try {
+    const {
+      username,
+      email,
+      password,
+      confirmPassword,
+      firstName,
+      middleName,
+      lastName,
+    } = req.body;
+  
+    if (
+      !username ||
+      !email ||
+      !password ||
+      !confirmPassword ||
+      !firstName ||
+      !lastName
+    ) {
+      return res
+        .status(400)
+        .json({ message: "All required fields must be filled out" });
+    }
+    if (password !== confirmPassword) {
+      return res.status(400).json({ message: "Passwords do not match" });
+    }
+  
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    if (!emailRegex.test(email)) {
+      return res.status(400).json({ message: "Invalid email format" });
+    }
+  
+    const hashedPassword = await bcrypt.hash(password, 10);
+  
+    UserInformation.createUserInformation(
+      firstName,
+      middleName,
+      lastName,
+      (err, result) => {
+        if (err) {
+          console.error("Error inserting into user_information table:", err);
+          return res.status(500).json({ message: "Internal server error" });
+        }
+  
+        const userInformationId = result.insertId;
+  
+        User.createUser(
+          userInformationId,
+          username,
+          email,
+          hashedPassword,
+          (err, result) => {
+            if (err) {
+              console.error("Error inserting into user table:", err);
+              return res.status(500).json({ message: "Internal server error" });
+            }
+  
+            res.status(201).json({ message: "User registered successfully" });
+          }
+        );
+      }
+    );
+
+  } catch (error) {
+    console.log(error);
```

Comment:
### Remove console.logs() and unneccessary codes in the production environment.
- Use console.error() for logging error messages.
- Remove codes/variables that are not used.
https://docs.google.com/presentation/d/1iCoz17mBxZtIQ9PifAW0jjKsuv4xDTlx-TPQxdLySow/edit#slide=id.g2637aafa621_0_0